docs/_docs/tech/steps.md

```diff
@@ -623,6 +613,169 @@ Column names must not be escaped. Strings have to be escaped with quotes.
+    ['currentDatasetColumn2', 'rightDatasetColumn2'],
+  ]
+}
+```
+
+#### Example 1: Left join with one columns couple as `on` parameter
+
+**Input dataset:**
+
+| Label   | Value |
+| ------- | ----- |
+| Label 1 | 13    |
+| Label 2 | 7     |
+| Label 3 | 20    |
+| Label 4 | 1     |
+| Label 5 | 1     |
+| Label 6 | 1     |
+
+**rightDataset (saved in the application Vuex store)**:
+
+| Label   | Group   |
+| ------- | ------- |
+| Label 1 | Group 1 |
+| Label 2 | Group 1 |
+| Label 3 | Group 2 |
+| Label 4 | Group 2 |
+
+**Step configuration:**
+
+```javascript
+{
+  name: 'join',
+  right_pipeline: 'rightDataset',
+  type: 'left',
+  on: [['Label', 'Label']];
+}
+```
+
+**Output dataset:**
+
+| Label   | Value | Group   |
+| ------- | ----- | ------- |
+| Label 1 | 13    | Group 1 |
+| Label 2 | 7     | Group 1 |
+| Label 3 | 20    | Group 2 |
+| Label 4 | 1     | Group 2 |
+| Label 5 | 1     |         |
+| Label 6 | 1     |         |
+
+#### Example 2: inner join with different columns names in the `on` parameter
+
+**Input dataset:**
+
+| Label   | Value |
+| ------- | ----- |
+| Label 1 | 13    |
+| Label 2 | 7     |
+| Label 3 | 20    |
+| Label 4 | 1     |
+| Label 5 | 1     |
+| Label 6 | 1     |
+
+**rightDataset (saved in the application Vuex store)**:
+
+| LabelRight | Group   |
+| ---------- | ------- |
+| Label 1    | Group 1 |
+| Label 2    | Group 1 |
+| Label 3    | Group 2 |
+| Label 4    | Group 2 |
+
+**Step configuration:**
+
+```javascript
+{
+  name: 'join',
+  right_pipeline: 'rightDataset',
+  type: 'inner',
+  on: [['Label', 'LabelRight']];
+}
+```
+
+**Output dataset:**
+
+| Label   | Value | LabelRight | Group   |
+| ------- | ----- | ---------- | ------- |
+| Label 1 | 13    | Label 1    | Group 1 |
+| Label 2 | 7     | Label 2    | Group 1 |
+| Label 3 | 20    | Label 3    | Group 2 |
+| Label 4 | 1     | Label 4    | Group 2 |
+
+#### Example 3: left outer join with several columns couples as `on` parameter
+
+Note that a left outer join is useful when you have a dataset specifying rows to
+be excluded from your current dataset example.
+
+**Input dataset:**
+
+| Label   | Value | Country   |
+| ------- | ----- | --------- |
+| Label 1 | 13    | Country 1 |
+| Label 1 | 7     | Country 2 |
+| Label 2 | 20    | Country 1 |
+| Label 2 | 1     | Country 2 |
+| Label 3 | 1     | Country 1 |
+| Label 3 | 1     | Country 2 |
+
+**rightDataset (saved in the application Vuex store)**:
+
+| Label   | Country   |
+| ------- | --------- |
+| Label 1 | Country 1 |
+| Label 1 | Country 2 |
+| Label 2 | Country 1 |
+| Label 3 | Country 2 |
+
+**Step configuration:**
+
+```javascript
+{
+  name: 'join',
+  right_pipeline: 'rightDataset',
+  type: 'left outer',
+  on: [['Label', 'Label'], ['Country', 'Country']];
+}
+```
+
+**Output dataset:**
+
+| Label   | Value | Country   |
+| ------- | ----- | --------- |
+| Label 2 | 1     | Country 2 |
+| Label 3 | 1     | Country 1 |
+
```

Comment:
This is not a `left outer join`. A `left outer join` would be an union of all rows where unmatched rows would be filled with `null` values.